game/forms.py

```
@@ -2,30 +2,45 @@
+    def clean_username(self):
+        username = self.cleaned_data.get('username')
+        if username:
+            from django.contrib.auth import get_user_model
+            User = get_user_model()
+            
+            if User.objects.filter(username__iexact=username, is_active=True).exists():
+                raise ValidationError("A user with that username already exists.")
+                
+            lock_session = cache.get(f"otp_lock:username:{username.lower()}")
+            if lock_session and self.request and self.request.session.session_key != lock_session:
+                raise ValidationError("This username is temporarily reserved. Please choose another or try again later.")
```

Ruff (0.15.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)